File: code/nxt-integration/src/main/java/org/fbme/integration/nxt/importer/CoordinateShift.java
package org.fbme.integration.nxt.importer;

public class CoordinateShift {
    public int yDiff;
    private Integer yInput;
    private Integer yOutput;
    private final Integer xInput;
    private final Integer xOutput;

    public CoordinateShift(Integer xInput, Integer xOutput) {
        yDiff = 50;
        yInput = 50;
        yOutput = 50;
        this.xInput = -xInput;
        this.xOutput = xOutput;
    }

    public Integer getY(Type type) {
        return type == Type.INPUT ? yInput : yOutput;
    }

    public Integer getX(Type type) {
        return type == Type.INPUT ? xInput : xOutput;
    }

    public void setY(Type type, Integer value) {
        if (type == Type.INPUT) {
            yInput = value;
        } else {
            yOutput = value;
        }
    }

    public void adjustYForParameters() {
        yInput = Math.max(yInput, yOutput) + 2 * yDiff;
        yOutput = yInput;
    }
}

File: code/nxt-integration/src/main/kotlin/org/fbme/integration/nxt/importer/CoordinateShift.kt
package org.fbme.integration.nxt.importer;

public class CoordinateShift {
    public int yDiff;
    private int yInput;
    private int yOutput;
    private final int xInput;
    private final int xOutput;

    public CoordinateShift(int xInput, int xOutput) {
        yDiff = 50;
        yInput = 50;
        yOutput = 50;
        this.xInput = -xInput;
        this.xOutput = xOutput;
    }

    public int getY(Type type) {
        return type == Type.INPUT ? yInput : yOutput;
    }

    public int getX(Type type) {
        return type == Type.INPUT ? xInput : xOutput;
    }

    public void setY(Type type, int value) {
        if (type == Type.INPUT) {
            yInput = value;
        } else {
            yOutput = value;
        }
    }

    public void adjustYForParameters() {
        yInput = Math.max(yInput, yOutput) + 2 * yDiff;
        yOutput = yInput;
    }
}
